### ISSUE TITLE
Feature request: notation scopes in "Tactic Notation"

### ISSUE THREAD

**Issue by rlepigre** [coq/coq#15793]
There is currently no way to give notation scopes to open for arguments of a `Tactic Notation`, which would be convenient in some cases. For example, this comes up in Iris where we end up having to writing things like `iAssert (...)%I as ...` all the time, whereas we would like to be able to just write `iAssert (...) as ...`. It would be great if we could to define `iAssert` with something like:
```coq
Tactic Notation "iAssert" open_constr(Q%I) "as" constr(pat) := ...
```

cc @Janno @RalfJung @robbertkrebbers

**Comment by RalfJung** [comment]
Duplicate of https://github.com/coq/coq/issues/4381 ?

**Comment by rlepigre** [comment]
Indeed, I did not think to search for closed issues, thanks @RalfJung. However, I still think this would be a useful feature to have.

**Comment by SkySkimmer** [comment]
Available in ltac2. Probably won't be added in ltac1.

**Comment by robbertkrebbers** [comment]
I know that this request has come up many times, and I remain puzzled why it shouldn't be added to Ltac1. Still >99% of the code between `Proof.` and `Qed` is written in Ltac1. So even if one were to write a tactic internally in Ltac2, one still needs an Ltac1 wrapper to make it usable.

**Comment by robbertkrebbers** [comment]
For the concrete use case of @rlepigre, this issue comes up all the time. For example, a student wrote the following in the evaluation of my course:

> We sometimes felt like we needed to fully grasp the internals of
Iris to work with it effectively. As an example: we are sometimes required to add %I after a statement to make sure we didn’t get errors. We found in the documentation that this evaluates notations in bi scope. However, this didn’t give us an intuition for when we would need %I, so we ended up using trial and error to figure out when to use it.

Without the feature requested in this issue there is no way for the Iris devs to fix this. (And saying that all endusers of Iris should use Ltac2 simply does not make sense.)
